lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -167,6 +167,14 @@ class BasicEntityPersister implements EntityPersister
      */
     private $insertSql;
 
+    /**
+     * List of columns of the currently cached insertSql.
+     * The insertSql is reset if another column list is given which happens e.g. when changing the id generator.
+     *
+     * @var string[]
```

Comment:
There's still a problem about the state of this: this is `string[]|null`, as otherwise line 1403 is controversial (checking against `null`)

Reply:
@Ocramius I changed the typehint for `$this->insertSql` as well to be consistent.